src/wei/routers/admin_routes.py

```diff
@@ -57,6 +62,14 @@ def reset_module(module_name: str) -> None:
     send_reset(state_manager.get_module(module_name))
 
 
+@router.api_route("/reset_wf/{wf_run_id}", methods=["POST"])
+def reset_workflow(wf_run_id: str) -> None:
+    """Restarts a workflow"""
+    state_manager.paused = True
```

Comment:
Not sure I understand why we're wrapping the commands by pausing/unpausing the workcell. Should the state manager lock be acquired out here instead of in the workflow.py methods?

Reply:
To be more clear: this seems dangerous because you can get into race conditions where multiple admin commands being sent can cause the workcell to pause/unpause mid-admin action